atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -444,27 +444,32 @@ public InetSocketAddress getRandomHostForKey(byte[] key) {
 
     public void addPool(InetSocketAddress server) {
         int currentPoolNumber = cassandraHosts.indexOf(server) + 1;
-        currentPools.put(
+        addPoolInternal(
                 server,
                 new CassandraClientPoolingContainer(metricsManager, server, config, currentPoolNumber, poolMetrics));
```

Comment:
Question: how does the whole `currentPoolNumber` work when you take hosts out and return them?
Answer: I _think_ that A) it's fine given that we're indexing this from the host in the list, not necessarily the number of hosts in the pool, and B) it's primarily for metrics